CodeEdit/Features/Settings/Pages/SourceControlSettings/SourceControlGitView.swift

```diff
@@ -24,23 +29,61 @@ struct SourceControlGitView: View {
                 showMergeCommitsInPerFileLog
             }
         }
+        .onAppear {
+            Task {
+                authorName = try await gitConfig.get(key: "user.name", global: true) ?? ""
+                authorEmail = try await gitConfig.get(key: "user.email", global: true) ?? ""
+                preferRebaseWhenPulling = try await gitConfig.get(key: "pull.rebase", global: true) ?? false
+                Task {
```

Comment:
This Task is nested inside another Task

Reply:
The Toggle onChange was getting fired onAppear even though hasAppear was set to true after these other variables were set. I think it was a timing issue. Putting it in Task fixed it.